LICENSE

```diff
@@ -199,3 +199,31 @@
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
+
+========================================================================
```

Comment:
After reviewing this further, I'm wondering whether it's better to add this block to the NOTICE file and leave the LICENSE file unchanged. @zabetak, any thoughts?

Reply:
the XTable NOTICE should be updated to include the main details from the Avro and Hudi NOTICE files.

Reply:
the changes in this PR seem unrelated to the xtable-utilities fat jar and seem more related to the XTable source code (and are needed - but the title of this PR seems wrong).

You also need a separate LICENSE and NOTICE for the xtable-utilities fat jar that list all the license and notice details of the classes that you include in that jar. There is a suspicion that some of the classes should not be in there because they have licenses that are not compatible with the Apache license. The OpenJDK JOL classes would seem to me that they should be removed because they are GPL licensed. https://github.com/openjdk/jol/blob/master/LICENSE

Reply:
Yes these changes are related to XTable source code and updated the PR title as well. There is an in-progress PR to exclude the category X dependencies from xtable-utilities bundle. 
https://github.com/apache/incubator-xtable/pull/538 

For release 0.2.0, we will be excluding the utilities bundle from the release.

Reply:
Although this PR is merged, I will comment here to keep the discussion in one place.

@jcamachor As a rule of thumb updating the LICENSE file is more frequent than updating the NOTICE file. In most cases we shouldn't touch the NOTICE file.

My understanding after reading https://infra.apache.org/licensing-howto.html#mod-notice is that in this case the NOTICE file does not need to be modified. In fact, we should strive to keep the NOTICE file very simple and "Do not add anything to NOTICE which is not legally required."

Other similar LEGAL tickets (e.g., LEGAL-234) imply that we don't need to duplicate the NOTICE file when we bundle code from other ASF projects.

@pjfanning Can you elaborate a bit on why you believe that the NOTICE file should be copied? 

For the LICENSE, file we are in a similar situation. According to https://infra.apache.org/licensing-howto.html#alv2-dep it seems that we don't necessarily need to add anything in the file. 

Finally, for the code adapted from Apache Hudi and Apache Spark I would argue that it does not classify as [3rd-party work](https://www.apache.org/legal/src-headers.html#3party) since the files are developed under the same legal entity (The Apache Software Foundation). These files are already submitted to the ASF by the copyright owner. Moving them around (even across projects) and modifying them does not consist in re-submission.

Anyways, I am not a legal expert but I would say that for the two classes that we adapted from Spark and Hudi we don't need anything in the LICENSE and NOTICE file.

@vinishjail97 Let's converge on this discussion before creating the RC2.

Reply:
@zabetak this appears in https://infra.apache.org/licensing-howto.html#alv2-dep
```
though the ASF copyright line and any other portions of NOTICE must be considered for propagation
```

Reply:
I agree that we don't need to duplicate this part.

```
This product includes software developed at
  The Apache Software Foundation (http://www.apache.org/).
```

Reply:
It says that it "must be considered for..." not it "must be propagated". The way I interpret this is that we should check what is there and if it affects the code we are touching do the necessary work.

Reply:
Have a look at https://github.com/apache/hudi/blob/master/NOTICE

This may not be correct but most ASF projects err on the including too much detail in the NOTICE. You get into less trouble for having too much detail than too little.

Reply:
Both too much and too little is problematic and adds burden to consumers. ASF projects copy the one another so we should not allow "bad" patterns to propagate.

Here we are talking about just two files so if there are doubts about what needs to be done let's open a LEGAL ticket.

Reply:
I'm not against removing the changes to the NOTICE file. The LICENSE changes were the most important part.

Reply:
I also think that the LICENSE changes are redundant.

Reply:
create a LEGAL issue then - most ASF projects include details of borrowed code from other ASF projects in their LICENSEs.